### PR TITLE
explicit double spending check

### PR DESCRIPTION
**Pull request by adecaro**
FSC must be updated to return a message about the kind of error that happened.

This PR depends on https://github.com/hyperledger-labs/fabric-smart-client/pull/353 

Signed-off-by: Angelo De Caro <adc@zurich.ibm.com>

### CODE REVIEW

File: integration/token/fungible/tests.go [hunk 1]
```diff
@@ -730,6 +741,44 @@ func TransferCash(network *integration.Infrastructure, id string, wallet string,
...
+		Recipient: network.Identity(receiver),
+	}))
+	if len(errorMsgs) == 0 {
+		Expect(err).NotTo(HaveOccurred())
+	} else {
+		Expect(err).To(HaveOccurred())
+		for _, msg := range errorMsgs {
+			Expect(err.Error()).To(ContainSubstring(msg))
+		}
+		time.Sleep(5 * time.Second)
```
Comment (yacovm) [inline integration/token/fungible/tests.go:783]: why the sleep?

Comment (adecaro) [reply integration/token/fungible/tests.go:783]: I don't remember to be honest. I see it also in the other functions. I can submit another PR where I remove all these sleeps and see what happens. Wdyt?

File: Makefile [hunk 2]
```diff
@@ -95,7 +95,7 @@ integration-tests-interop-dlog:
 
 .PHONY: tidy
 tidy:
-	@go mod tidy -compat=1.17
```
[the remark this answers is not in the document]
Comment (adecaro) [reply Makefile:103]: sorry, a mistake

File: integration/token/fungible/views/auditor.go [hunk 3]
```diff
@@ -187,7 +186,7 @@ func (r *CurrentHoldingView) Call(context view.Context) (interface{}, error) {
 	filter, err := aqe.NewHoldingsFilter().ByEnrollmentId(r.EnrollmentID).ByType(r.TokenType).Execute()
 	assert.NoError(err, "failed retrieving holding for [%s][%s]", r.EnrollmentID, r.TokenType)
 	currentHolding := filter.Sum()
-	decimal := currentHolding.Decimal()
+	decimal := currentHolding.Text(10)
```
[the remark this answers is not in the document]
Comment (adecaro) [reply integration/token/fungible/views/auditor.go:189]: currentHolding  is a big.Int and `Text` is a function of big.Int. What's not intuitive?

File: integration/token/fungible/tests.go [hunk 4]
```diff
@@ -730,6 +764,46 @@ func TransferCash(network *integration.Infrastructure, id string, wallet string,
...
+	return res.TxID, res.TXRaw
+}
+
+func BroadcastPreparedTransferCash(network *integration.Infrastructure, id string, tx []byte, finality bool, errorMsgs ...string) {
+	_, err := network.Client(id).CallView("broadcastPreparedTransfer", common.JSONMarshall(&views.BroadcastPreparedTransfer{
+		Tx:       tx,
+		Finality: finality,
+	}))
+	if len(errorMsgs) == 0 {
+		Expect(err).NotTo(HaveOccurred())
```
[the remark this answers is not in the document]
Comment (adecaro) [reply integration/token/fungible/tests.go:796]: done

File: integration/token/fungible/tests.go [hunk 5]
```diff
@@ -730,6 +764,46 @@ func TransferCash(network *integration.Infrastructure, id string, wallet string,
...
+	common.JSONUnmarshal(txBoxed.([]byte), res)
+	return res.TxID, res.TXRaw
+}
+
+func BroadcastPreparedTransferCash(network *integration.Infrastructure, id string, tx []byte, finality bool, errorMsgs ...string) {
+	_, err := network.Client(id).CallView("broadcastPreparedTransfer", common.JSONMarshall(&views.BroadcastPreparedTransfer{
+		Tx:       tx,
+		Finality: finality,
+	}))
+	if len(errorMsgs) == 0 {
```
[the remark this answers is not in the document]
Comment (adecaro) [reply integration/token/fungible/tests.go]: done, everywhere!

**Review by yacovm — APPROVED** [review]

**Pull request merged**